tests/phpunit/tests/kses/wpFilterGlobalStylesPost.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * @group kses
+ *
+ * @covers ::wp_filter_global_styles_post
+ */
+class Tests_Kses_WpFilterGlobalStylesPost extends WP_UnitTestCase {
+
+	/**
+	 * Theme data.
+	 *
+	 * @var array
+	 */
+	private $user_theme_data = array(
+		'isGlobalStylesUserThemeJSON' => 1,
+		'version'                     => 1,
+		'nonSchemaRule'               => 'someValue',
+		'styles'                      => array(
+			'blocks' => array(
+				'core/button' => array(
+					'border' => array(
+						'radius' => '0',
+					),
+				),
+			),
+		),
+	);
```

Comment:
I'm thinking a data provider is needed to have more datasets for better coverage. There are a lot of schema properties to test. What do you all think?

Reply:
I've added a new data provider for the `test_should_not_remove_safe_global_style_rules`.